src/frontend/src/components/dock.js

```diff
@@ -122,8 +123,22 @@ class Dock extends PureComponent<Props> {
       }
     }
 
-    profile.favorites = favorites;
-    handleAuthChange(profile);
+    // Save updated profile
+    const updatedProfile = { ...profile, favorites };
```

Comment:
This makes a copy of the profile so that original object isn't changed by this function. This prevents the favorite from showing up until it's saved successfully.